robot/web/web.go

```diff
@@ -777,6 +783,46 @@ func (svc *webService) initAuthHandlers(listenerTCPAddr *net.TCPAddr, options we
 	return rpcOpts, nil
 }
 
+func parseLegacyAPIKeys(handler config.AuthHandlerConfig) []string {
+	apiKeys := handler.Config.StringSlice("keys")
+	if len(apiKeys) == 0 {
+		apiKey := handler.Config.String("key")
+		if apiKey == "" {
+			return []string{}
+		}
+		apiKeys = []string{apiKey}
+	}
+
+	return apiKeys
+}
+
+func parseAPIKeys(handler config.AuthHandlerConfig) map[string]string {
+	apiKeys := map[string]string{}
+	for k := range handler.Config {
+		// if it is not a legacy api key indicated by "key(s)" key
+		// current api keys will follow format { [keyId]: [key] }
+		if k != "keys" && k != "key" {
+			apiKeys[k] = handler.Config.String(k)
+		}
+	}
+	return apiKeys
+}
+
+// makeMultiStepAPIKeyAuthHandler supports auth handlers for both legacy and non-legacy api keys for backwards compatibility.
+func makeMultiStepAPIKeyAuthHandler(legacyEntities, legacyExpectedPayloads []string, apiKeys map[string]string) rpc.AuthHandler {
```

Comment:
to confirm: here, legacyEntities = the robot api keys previous to the initiation of the api keys project?

Reply:
so legacyEntities = robot fqdns. previously, we matched robot fqds to legacy api keys (aka legacyExpectedPayloads)

Reply:
i can update these param names to be more specific though since we are in an auth handler specific to API keys now. thats a good callout. anything to make this chunk of code less confusing 🤣

Reply:
i actually still think legacyEntities makes sense here still but updated to be `legacyExpectedAPIKeys`. let me know what you think

Reply:
ok gotcha, since legacy api keys still exist. sgtm!